tests/test_utilities.py

```diff
@@ -116,3 +116,21 @@ def test_flatten_dict():
     inp = {"a": {"b": 1, "c": 2}, "d": 3}
     out = _flatten_dict(inp)
     assert out == {"b": 1, "c": 2, "d": 3}
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="test requires gpu")
+def test_bincount():
+    """test that bincount works in deterministic setting on GPU."""
+    torch.use_deterministic_algorithms(True)
+
+    x = torch.randint(100, size=(100,))
+    # uses custom implementation
+    res1 = _bincount(x, minlength=10)
+
+    torch.use_deterministic_algorithms(False)
+
+    # uses torch.bincount
+    res2 = _bincount(x, minlength=10)
+
+    # check for correctness
+    assert torch.allclose(res1, res2)
```

Comment:
should this be rather identical?

Reply:
pytorch recommends using torch.allclose especially when dealing with floating point tensors. But once we changed the default dtype to torch.long, this should be the same. So I recommend keeping it like this, since with integer-tensors you cannot reach the specified tolerances without them being identical :)

Reply:
agree here :]

Reply:
well but the counts are `dtype=torch.long`, right?

Reply:
weel the code below has it set as `dtype=torch.long`, right?